app/src/main/java/com/tpp/theperiodpurse/ui/calendar/CalendarScreen.kt

```diff
@@ -58,6 +65,114 @@ val tabModifier = Modifier
+                symptoms = trackedSymptoms,
+                onSymptomOnClicks = trackedSymptoms.map { { activeSymptom = it } },
+                modifier = tabModifier
+            )
+        }
+    }
+}
+
+@Composable
+private fun DisplaySymptomTab(
+    activeSymptom: Symptom?,
+    expanded: Boolean,
+    onExpandButtonClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.Center,
+        modifier = modifier
+    ) {
+        Text(
+            text = stringResource(activeSymptom?.nameId ?: Symptom.FLOW.nameId),
+            fontWeight = FontWeight.Bold,
+            modifier = Modifier.padding(end = 2.dp)
+        )
+        Icon(
+            painter = painterResource(
+                id = activeSymptom?.resourceId ?: Symptom.FLOW.resourceId
+            ),
+            tint = Color.Black,
+            contentDescription = activeSymptom?.name,
+            modifier = Modifier
+                .padding(end = 0.dp)
+                .testTag("Selected Symptom")
+        )
+        SwitchSymptomButton(
+            expanded = expanded,
+            onClick = onExpandButtonClick
+        )
+    }
+}
+
+@Composable
+private fun SwitchSymptomButton(
+    expanded: Boolean,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    IconButton(onClick = onClick, modifier = modifier) {
+        Icon(
+            imageVector = if (expanded) Icons.Filled.ExpandLess else Icons.Filled.ExpandMore,
+            tint = Color.Gray,
+            contentDescription = stringResource(R.string.expand_button_symptoms_content_description)
+        )
+    }
+}
+
+@Composable
+private fun SwitchSymptomTab(
+    activeSymptom: Symptom?,
+    symptoms: List<Symptom>,
+    onSymptomOnClicks: List<() -> Unit>,
+    modifier: Modifier = Modifier
+) {
+    Row(
+        horizontalArrangement = Arrangement.SpaceEvenly,
+        modifier = modifier.testTag("Symptom Options")
+    ) {
+        symptoms.zip(onSymptomOnClicks).forEach { (symptom, onClick) ->
+            val interactionSource = remember { MutableInteractionSource() }
+            val isPressed by interactionSource.collectIsPressedAsState()
+
+            IconButton(
+                onClick = onClick,
+                interactionSource = interactionSource
+            ) {
+                val defaultColor = Color.Black
+                val color = if (isPressed) {
+                    defaultColor.copy(ContentAlpha.disabled)
+                } else if (activeSymptom == symptom) {
+                    Color(0xFFBF3428)
+                } else {
+                    defaultColor
+                }
+                Icon(
+                    painter = painterResource(id = symptom.resourceId),
+                    tint = color,
+                    contentDescription = stringResource(symptom.nameId)
+                )
+            }
+        }
+    }
+}
```

Comment:
Are these duplicated from `calendar/components/SymptomTab.kt` ?

Reply:
They're not direct copy & paste but they're similar. I believe SymptomTab has more to its functions than the functions used in CalendarScreen. @michaelng0107 is it safe to remove these duplicate functions from CalendaryScreen?